docs/spec/auth/scope.md

```diff
@@ -56,7 +56,7 @@ it.
 The resource name represent the name which identifies a resource for a resource
 provider. A resource is identified by this name and the provided resource type.
 An example of a resource name would be the name component of an image tag, such
-as "samalba/myapp".
+as "samalba/myapp" or "hostname/samalba/myapp".
```

Comment:
It is confusing and ambiguous but don't see it as a problem here. From the scope perspective it only cares about what the resource name is, it is up to the token server to decide whether it is a canonical name or not, as well as whether it has something that matches. Every token server may have slightly different rules, such as with the Docker hub resource names always have a username and repository name making it unambiguous. We will need to figure out though if for private registries which are attempting to use the token server for both a registry which is using non-canonical names and a notary server which is using canonical names, whether there will be ambiguity. If there is ambiguity, we will probably need to solve it by adding a flag or header rather than by attempting to clarify in the grammar.